packages/kit/src/core/generate_manifest/index.js

```diff
@@ -111,14 +111,32 @@ export function generate_manifest({ build_data, relative_path, routes, format =
 					types: ${s(route.types)},
 					optional: ${s(route.optional)},
 					page: ${route.page ? `{ layouts: ${get_nodes(route.page.layouts)}, errors: ${get_nodes(route.page.errors)}, leaf: ${reindexed.get(route.page.leaf)} }` : 'null'},
-					endpoint: ${route.endpoint ? loader(`${relative_path}/${resolve_symlinks(build_data.server.vite_manifest, route.endpoint.file).chunk.file}`) : 'null'}
+					endpoint: ${route.endpoint ? loader(joinUrlSegments(relative_path, resolve_symlinks(build_data.server.vite_manifest, route.endpoint.file).chunk.file)) : 'null'}
 				}`;
 				}).filter(Boolean).join(',\n\t\t\t\t')}
 			],
 			matchers: async () => {
-				${Array.from(matchers).map(type => `const { match: ${type} } = await ${load(`${relative_path}/entries/matchers/${type}.js`)}`).join('\n\t\t\t\t')}
+				${Array.from(matchers).map(type => `const { match: ${type} } = await ${load(joinUrlSegments(relative_path, `/entries/matchers/${type}.js`))}`).join('\n\t\t\t\t')}
 				return { ${Array.from(matchers).join(', ')} };
 			}
 		}
 	}`.replace(/^\t/gm, '');
 }
+
+/**
+ * @param {string} a
+ * @param {string} b
+ * @returns
+ */
+export function joinUrlSegments(a, b) {
```

Comment:
This is an internal API, so it should use snake case. I also don't think it needs to be exported from this module.

Reply:
yep, thanks. I swiped this from Vite and forgot to update it :smile: